src/components/app/AppLocalizationProvider.js

```diff
@@ -2,59 +2,52 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 // @flow
-
-import { negotiateLanguages } from '@fluent/langneg';
-import { FluentBundle, FluentResource } from '@fluent/bundle';
-import { LocalizationProvider, ReactLocalization } from '@fluent/react';
+import explicitConnect from 'firefox-profiler/utils/connect';
+import { LocalizationProvider } from '@fluent/react';
 import * as React from 'react';
+import type { L10nState } from 'firefox-profiler/types';
+import { setupLocalization } from 'firefox-profiler/actions/l10n';
+import type { ConnectedProps } from 'firefox-profiler/utils/connect';
+import { getL10nState } from 'firefox-profiler/selectors/app';
 
-// Store all translations as a simple object which is available
-// synchronously and bundled with the rest of the code.
-
-type Props = {|
+type StateProps = {|
+  +l10nState: L10nState,
+|};
+type OwnProps = {|
   children: React.Node,
 |};
+type DispatchProps = {|
+  +setupLocalization: typeof setupLocalization,
+|};
 
-const RESOURCES = {
-  fr: new FluentResource(
-    "intro = &mdash; Application Web pour l'analyse des performances de Firefox"
-  ),
-  'en-US': new FluentResource(
-    'intro = &mdash; Web app for Firefox performance analysis'
-  ),
-  pl: new FluentResource(
-    'intro = Aplikacja internetowa do analizy wydajności przeglądarki Firefox'
-  ),
-};
-
-// A generator function responsible for building the sequence
-// of FluentBundle instances in the order of user's language
-// preferences.
-function* generateBundles(userLocales) {
-  // Choose locales that are best for the user.
-  const currentLocales = negotiateLanguages(
-    userLocales,
-    ['fr', 'en-US', 'pl'],
-    {
-      defaultLocale: 'en-US',
-    }
-  );
+type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
 
-  for (const locale of currentLocales) {
-    const bundle = new FluentBundle(locale);
-    bundle.addResource(RESOURCES[locale]);
-    yield bundle;
+class AppLocalizationProviderImpl extends React.PureComponent<Props> {
+  componentDidMount() {
+    const { setupLocalization } = this.props;
+    const locales = navigator.languages;
+    setupLocalization(locales);
   }
-}
 
-// The ReactLocalization instance stores and caches the sequence of generated
-// bundles. You can store it in your app's state.
-const l10n = new ReactLocalization(generateBundles(navigator.languages));
-
-export class AppLocalizationProvider extends React.PureComponent<Props> {
   render() {
-    const { children } = this.props;
-
-    return <LocalizationProvider l10n={l10n}>{children}</LocalizationProvider>;
+    const { children, l10nState } = this.props;
+    return (
+      <LocalizationProvider l10n={l10nState.localization}>
+        {children}
+      </LocalizationProvider>
+    );
   }
 }
+export const AppLocalizationProvider = explicitConnect<
+  OwnProps,
+  StateProps,
+  DispatchProps
+>({
+  mapStateToProps: state => ({
+    l10nState: getL10nState(state),
```

Comment:
We are getting the whole l10n state in this component, but we are only using `l10nState.localization`. We can directly get the localization with `getLocalizations` selector instead.